plaid/src/config.rs

```diff
@@ -183,24 +184,46 @@ pub async fn configure() -> Result<Configuration, ConfigurationError> {
                 .long("config")
                 .default_value("./plaid/resources/plaid.toml")
                 .takes_value(true),
+        )
+        .arg(
+            Arg::new("secrets")
+                .help("Path to the secrets json file")
+                .long("secrets")
+                .default_value("./plaid/private-resources/secrets.json")
+                .takes_value(true),
         ).get_matches();
 
     // Read the configuration file
-    let config = match tokio::fs::read(matches.value_of("config").unwrap()).await {
+    let mut config = match tokio::fs::read_to_string(matches.value_of("config").unwrap()).await {
         Ok(config) => config,
         Err(e) => {
-            println!("Encountered file error when trying to read configuration!");
-            println!("{}", e);
+            error!("Encountered file error when trying to read configuration!. Error: {e}");
+            return Err(ConfigurationError::FileError);
+        }
+    };
+
+    // Read the secrets file and parse into a serde object
+    let secrets = match tokio::fs::read(matches.value_of("secrets").unwrap()).await {
+        Ok(secret_bytes) => {
+            let secrets = serde_json::from_slice::<Value>(&secret_bytes).unwrap();
+            secrets.as_object().cloned().unwrap()
+        }
+        Err(e) => {
+            error!("Encountered file error when trying to read secrets file!. Error: {e}");
             return Err(ConfigurationError::FileError);
         }
     };
 
+    // Iterate over the secrets we just parsed and replace matching keys in the config
+    for (secret, value) in secrets {
+        config = config.replace(&secret, value.as_str().unwrap());
```

Comment:
I'm worried this is too powerful for it's own good, but it's so flexible I'm tempted to keep it.

Not sure if this was your intention but this lets you hide entire part of the plaid config in the secrets.

Reply:
That was not the original intention but you're right that this allows you to replace literally anything you want

Reply:
So crazy it just might work